test/suites/clustering_move.sh

```diff
@@ -0,0 +1,198 @@
+test_clustering_move() {
+  # shellcheck disable=2039,3043,SC2034
+  local LXD_DIR
+
+  setup_clustering_bridge
+  prefix="lxd$$"
+  bridge="${prefix}"
+
+  setup_clustering_netns 1
+  LXD_ONE_DIR=$(mktemp -d -p "${TEST_DIR}" XXX)
+  chmod +x "${LXD_ONE_DIR}"
+  ns1="${prefix}1"
+  spawn_lxd_and_bootstrap_cluster "${ns1}" "${bridge}" "${LXD_ONE_DIR}"
+
+  # Add a newline at the end of each line. YAML as weird rules..
+  cert=$(sed ':a;N;$!ba;s/\n/\n\n/g' "${LXD_ONE_DIR}/cluster.crt")
+
+  # Spawn a second node
+  setup_clustering_netns 2
+  LXD_TWO_DIR=$(mktemp -d -p "${TEST_DIR}" XXX)
+  chmod +x "${LXD_TWO_DIR}"
+  ns2="${prefix}2"
+  spawn_lxd_and_join_cluster "${ns2}" "${bridge}" "${cert}" 2 1 "${LXD_TWO_DIR}"
+
+  # Spawn a third node
+  setup_clustering_netns 3
+  LXD_THREE_DIR=$(mktemp -d -p "${TEST_DIR}" XXX)
+  chmod +x "${LXD_THREE_DIR}"
+  ns3="${prefix}3"
+  spawn_lxd_and_join_cluster "${ns3}" "${bridge}" "${cert}" 3 1 "${LXD_THREE_DIR}"
+
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster set node2 scheduler.instance=group
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster set node3 scheduler.instance=manual
+
+  ensure_import_testimage
+
+  # Preparation
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster group create foobar1
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster group assign node1 foobar1,default
+
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster group create foobar2
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster group assign node2 foobar2,default
+
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster group create foobar3
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster group assign node3 foobar3,default
+
+  LXD_DIR="${LXD_ONE_DIR}" lxc init testimage c1 --target node1
+  LXD_DIR="${LXD_ONE_DIR}" lxc init testimage c2 --target node2
+  LXD_DIR="${LXD_ONE_DIR}" lxc init testimage c3 --target node3
+
+  # Perform default move tests falling back to the built in logic of choosing the node
+  # with the least number of instances when targeting a cluster group.
+
+  LXD_DIR="${LXD_ONE_DIR}" lxc move c1 --target node2
+  LXD_DIR="${LXD_ONE_DIR}" lxc move c1 --target @foobar1
+  LXD_DIR="${LXD_ONE_DIR}" lxc info c1 | grep -q "Location: node1"
+
+  # c1 cannot be moved to the default group since it is already a member
+  ! LXD_DIR="${LXD_ONE_DIR}" lxc move c1 --target=@default || false
```

Comment:
Is this new behaviour in this PR?

Reply:
Yes, I have added an explicit check for this. Otherwise  if a cluster group has multiple members, you can run `lxc move --target @group` infinite times and the instance will always be moved to another member (depending on the choosen scheduler configuration). Maybe it would be a good idea to add a hint in the docs?

Reply:
I'm not sure about this. One thing that may hint at a cause of confusion here is that I saw hints in some of the comments in this PR that an instance can be considered a "member" of a particular cluster group. I don't think that is really true.

Instances are not members of cluster groups. 

Instead when creating or moving an instance one can "prune" the potential candidates for the cluster member placement using a cluster group selector. So when thinking about it this way, as long as we always exclude the instance's current cluster member location from the list of candidates, then it could be totally valid to move an instance using the same cluster group multiple times.

Reply:
Mhh yes, valid use case. It could be kind of a rescue mechanism to move all instances away from a member but still having them inside the same group. Otherwise you would need to create another group excluding a specific cluster member.

I'll add another test case for this behavior and remove the check.

Reply:
Well we do have `lxc cluster evacuate` for that, but it could be useful as a form of manual rebalancing.

Reply:
Check removed and additional test case added.